vacuum-advisor/vacuum-simulator.mdx

```diff
@@ -0,0 +1,182 @@
+  the simulation may look a lot different than the actual data.
+* Autovacuums triggered by other causes usually try to freeze old rows.
+  The simulation also freezes them, however, there are cases that freezing won't
+  happen when there are nothing can be frozen, or autovacuum is more focused on
+  cleaning up dead rows.
+  This can cause the transaction age to be reset even when it shouldn't be in
+  the simulation, potentially cause the simulation to trigger less autovacuums
+  by freeze age.
+
+
+## Triggered by inserts
+
+Inserts autovacuum is introduced on Postgres 13.
+With an inserts only table, because it won't generate any dead rows, autovacuums
+won't be triggered by dead rows. It can still be triggered by freeze age, but
+depending on the database workload, it can be a long time until it's triggered.
+Inserts autovacuum is triggered to freeze old transaction IDs like freeze age
+autovacuum, but more importantly, it is triggered to update a visibility map.
+Each table has a visibility map to keep track of which pages contain only tuples
+that are known to be visible to all active transactions. For the case like an
+inserts only table, all the pages of the table should be visible here and less
+likely to be not visible moving forward (will be flagged as not visible if there
+is any write operation on that page).
+However, Postgres doesn't know about it until VACUUM runs to set a bit (flag).
+Updating the visibility map will help to spend less time for any subsequent
+VACUUMs as these pages don't need to be VACUUMed and will be skipped. Updating
+the visibility map also helps to use index-only scans, as this information will
+be used to determinate weather to use index-only scans or not.
+
+![Triggered by inserts](vacuum_simulator_inserts.png)
+
+When the number of inserted rows (since the last VACUUM) exceeds the following
+threshold, inserts autovacuums will be triggered:
+
+```
+threshold = autovacuum_vacuum_insert_threshold + autovacuum_vacuum_insert_scale_factor * number of tuples (pg_class.reltuples, number of live rows)
+```
+
+With VACUUM Simulator, you can tweak `autovacuum_vacuum_insert_threshold` and
+`autovacuum_vacuum_insert_scale_factor` in this section.
+Lowering these configuration settings triggers autovacuums more with some
+inserts only or inserts heavy table, though this rarely triggers autovacuums for
+other types of tables.
+
+If you are observing too many inserts autovacuums with the table, you can make
+`autovacuum_vacuum_insert_scale_factor` 0 and adjust
+`autovacuum_vacuum_insert_threshold` to find a reasonable threshold for inserts
+autovacuums.
+
+### Limitation
+
+* This simulation is only available with servers running Postgres 13 or above.
+* With the simulation, inserts autovacuum freezes old transaction IDs as it can
+  do it.
+  However, in reality, it's less likely that inserts autovacuum will freeze old
+  transaction IDs.
```

Comment:
```suggestion
* With the simulation, INSERT-based autovacuum freezes old transaction IDs as it can.
  However, in reality, it's less likely that it will freeze old transaction IDs.
```

Why less likely? I think this is also something we should try to explain better.

Reply:
I'm actually removing this, I wasn't understanding this behavior well enough -- freezing was skipped because the current oldest unfrozen transaction ID age was under `vacuum_freeze_min_age`. I'm planning to make a logic change in the simulator so we should be able to remove this limitation.